### PR TITLE
add two options inputStyle and nativeSpellcheck

### PR DESCRIPTION
**Pull request by firm1**
Add two options for enable native (browsers) spellcheck on contenteditable instead of codemirror spellcheck which are limited.

I can link this issue : #52

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,6 +180,8 @@ easyMDE.value('New input for **EasyMDE**');
 - **shortcuts**: Keyboard shortcuts associated with this instance. Defaults to the [array of shortcuts](#keyboard-shortcuts).
 - **showIcons**: An array of icon names to show. Can be used to show specific icons hidden by default without completely customizing the toolbar.
 - **spellChecker**: If set to `false`, disable the spell checker. Defaults to `true`.
+- **inputStyle**: `textarea` or `contenteditable`. Defaults to `textarea` for desktop and `contenteditable` for mobile. `contenteditable` option is necessarry for enable nativeSpellcheck.
```
Comment (Situphen) [inline README.md]:
```suggestion
- **inputStyle**: `textarea` or `contenteditable`. Defaults to `textarea` for desktop and `contenteditable` for mobile. `contenteditable` option is necessary to enable nativeSpellcheck.
```

File: src/js/easymde.js [hunk 2]
```diff
@@ -1858,6 +1858,8 @@ EasyMDE.prototype.render = function (el) {
         placeholder: options.placeholder || el.getAttribute('placeholder') || '',
         styleSelectedText: (options.styleSelectedText != undefined) ? options.styleSelectedText : !isMobile(),
         configureMouse: configureMouse,
+        inputStyle: (options.inputStyle != undefined) ? options.inputStyle : (!isMobile()) ? 'textarea' : 'contenteditable',
```
Comment (Situphen) [inline src/js/easymde.js]:
```suggestion
        inputStyle: (options.inputStyle != undefined) ? options.inputStyle : isMobile() ? 'contenteditable' : 'textarea',
```

**Comment by Ionaru** [comment]
Can you add this to the changelog as well?

**Comment by firm1** [comment]
@Ionaru and voila.

Cheers.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,9 +5,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `inputStyle` and `nativeSpellcheck` options to manage the native language of the browser. [#52]
```
Comment (Situphen) [inline CHANGELOG.md]:
```suggestion
- `inputStyle` and `nativeSpellcheck` options to manage the native language of the browser. (Thanks to [@firm1], [#52]). 
```

**Comment by CatalinFrancu** [comment]
Silly question, and I'm sorry if this is the wrong place to ask, but how does one use this? I call the constructor with `nativeSpellcheck: true` and `inputStyle: 'contenteditable'`, but nothing seems to change. If the field contents are in a language other than English, every word is highlighted. Furthermore, the browser still fetches `en_US.{aff,dic}` from a CDN.

The demo website, https://easymde.tk/, seems to be down at the moment. Perhaps you could add a `contenteditable` example under `example/`?

Thanks!

**Comment by A-312** [comment]
@CatalinFrancu `spellchecker: false,` https://github.com/zestedesavoir/zds-site/commit/7eadd7e4d430c7a5b844f9cbe716655b8ecb3d57#diff-ac76f1d75eb777f747e6708b19d31b78R390

**Comment by CatalinFrancu** [comment]
@A-312 Thank you. I tried that, but that disables the spell checker altogether, both the CodeMirror one and the browser one. (The browser spell checker does work in other fields on the same page). Should I replace the `textarea` with a `div`? Should I set `contenteditable="true"` on the HTML element?

**Comment by A-312** [comment]
You a have to use the tagged version `next` ?

Because the release version doesn't have this change yet.


https://github.com/zestedesavoir/zds-site/commit/7eadd7e4d430c7a5b844f9cbe716655b8ecb3d57#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R47

**Comment by CatalinFrancu** [comment]
Brilliant! It works perfectly. I don't use npm, so just in case it helps other people, I ran the command `npm i easymde@2.9.1-277.0` as described [here](https://www.npmjs.com/package/easymde/v/2.9.1-277.0), then I copied the resulting files from under `node_modules/easymde/dist` to my source tree.

**Pull request merged**